waku/waku_sync/raw_bindings.nim

```diff
@@ -152,29 +153,36 @@ proc insert*(storage: Storage, id: int64, hash: WakuMessageHash): Result[void, s
   var bufPtr = addr(buffer)
   let res = raw_insert(storage, uint64(id), bufPtr)
   
-  #TODO error handling
+  #TODO error handling once we move to lmdb
 
   if res:
     return ok()
   else:
     return err("insert error")
 
-proc new*(T: type Negentropy, storage: Storage, frameSizeLimit: uint64): T =
-  let negentropy = constructNegentropy(storage, frameSizeLimit)
-  
-  #TODO error handling
+proc `==` *(a : Negentropy, b : pointer):bool{.borrow.}
 
+proc new*(T: type Negentropy, storage: Storage, frameSizeLimit: uint64): T {.raises: [ValueError] .} =
+  let negentropy = constructNegentropy(storage, frameSizeLimit)
+  if negentropy == nil:
+    raise newException(ValueError, "negentropy initialization failed due to lower framesize")
```

Comment:
`: Result[T, string] = ` instead

Reply:
I used an exception as this will propagate way up to the app. Did not want to keep sending errors up functions.

Reply:
I'm not sure when it is appropriate to use exception in Nim. Rust only use `Result`

Reply:
I was looking at Store protocol implementation and just followed it.